nms/app/views/alarms/components/__tests__/FilterRouteByRuleName-test.ts

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright 2022 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AlertRoutingTree} from '../AlarmAPIType';
+import {filterRouteByRuleName, filterUpdatedFilterRoutes} from '../hooks';
+
+const response = {
+  receiver: 'test_tenant_base_route',
+  match: {networkID: 'test'},
+  routes: [
+    {receiver: 'User1', match: {alertname: 'High Disk Usage Alert'}},
+    {
+      receiver: 'User1',
+      match: {alertname: 'Certificate Expiring Soon'},
+    },
+    {
+      receiver: 'User1',
+      match: {alertname: 'Bootstrap Exception Alert'},
+    },
+    {
+      receiver: 'User2',
+      match: {alertname: 'Gateway Checkin Failure'},
+    },
+  ],
+};
+
+describe('Testing filterRouteByRuleName', () => {
+  test('Remove only  High Disk Usage Alert from User1', () => {
+    const ruleName = 'High Disk Usage Alert';
+    const initialReceiver = 'User1';
+    const routes = filterRouteByRuleName(response, initialReceiver, ruleName);
+    expect(routes.length).toBe(3);
+  });
+
+  test('Remove only  Gateway Checkin Failure from User2', () => {
+    const ruleName = 'Gateway Checkin Failure';
+    const initialReceiver = 'User2';
+    const routes = filterRouteByRuleName(response, initialReceiver, ruleName);
+    expect(routes.length).toBe(3);
+  });
+
+  test('return all routes if initialReceiver is empty', () => {
```

Comment:
E.g. 2: Capitalization is different from above. It should be consistent. More below.